src/spicy/spicyz/glue-compiler.cc

```diff
@@ -1194,6 +1194,23 @@ struct VisitorZeekType : hilti::visitor::PreOrder<hilti::Result<hilti::Expressio
         return (! ids.empty() && ids.front().has_value()) ? ids.front()->namespace_() : hilti::ID();
     }
 
+    result_t create_record_type(const hilti::ID& ns, const hilti::ID& local,
+                                const std::vector<hilti::Expression>& fields) {
+        if ( fields.size() ) {
```

Comment:
The `if`/`else` branches are only for logging and can do quite some work just iterating, even if logging is disabled.

WDYT about wrapping all of this in a `if ( hilti::logger().isEnabled(ZeekPlugin) ) {
...`?